RoddyCore/src/de/dkfz/roddy/plugins/PluginInfo.java

```diff
@@ -99,21 +104,25 @@ public PluginInfo(String name, File zipFile, File directory, File developmentDir
 
     protected void fillListOfToolDirectories() {
         File toolsBaseDir = null;
-        try {
-            toolsBaseDir = getToolsDirectory();
+        toolsBaseDir = getToolsDirectory();
 
-            if (toolsBaseDir != null && toolsBaseDir.exists() && toolsBaseDir.isDirectory()) { //Search through the default folders, if possible.
+        if (toolsBaseDir != null && toolsBaseDir.exists() && toolsBaseDir.isDirectory()) { //Search through the default folders, if possible.
                 for (File file : toolsBaseDir.listFiles()) {
-                    if (!file.isDirectory() || file.isHidden()) {
-                        errors.add("Ignore directory " + file.getAbsolutePath() + "; It is not a valid tools directory.");
+                    PosixFileAttributes attr;
+                    try {
+                        attr = Files.readAttributes(file.toPath(), PosixFileAttributes.class);
+                    } catch (IOException ex) {
+                        errors.add("An IOException occurred while accessing '" + file.getAbsolutePath() + "': " + ex.getMessage());
+                        continue;
+                    }
+
+                    if (attr.isRegularFile() || file.isHidden()) {
```

Comment:
I don't know if this check is valid or behaves like it should. The former code checked, if "file" is  not a directory. Here you turn it around and check if it is a regular file. What about links etc.?

Reply:
Yes, I'll change it back. For some reason I remembered (incorrectly) that the access via PosixFileAttributes dereferences a symlink. But that is obviously wrong.